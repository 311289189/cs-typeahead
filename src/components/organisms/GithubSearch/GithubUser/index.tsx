import React from 'react';
import classnames from 'classnames'

import { User } from "../../../../../types/github";
const styles = require('./styles.module.css')

interface Props {
  user: User
  isLoading: boolean
}

export default ({ user, isLoading }: Props) => (
  <a
    href={user.html_url}
    target="_blank"
    rel="noopener noreferrer"
    className={classnames(styles.userTile, { [styles.loading]: isLoading })}
  >
    <img
      className={styles.avatar}
      src={user.avatar_url}
      alt={user.login}
    />
    <div className={styles.userLogin}>{user.login}</div>
  </a>
)
