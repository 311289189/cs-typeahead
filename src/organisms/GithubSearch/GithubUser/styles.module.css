.loading {
    opacity: 0.5;
}

.userTile {
    display: flex;
    padding: 10px;
    text-decoration: none;
}

.userTile:hover {
    background: #e1eaf6;
}

.avatar {
    width: 50px;
    height: 50px;
}

.userLogin {
    margin-left: 10px;
}
