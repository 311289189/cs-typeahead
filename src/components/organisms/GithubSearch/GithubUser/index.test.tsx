import React from 'react';
import renderer from 'react-test-renderer';

import GithubUser from "./index";

const mockUserFixture = {
  login: 'string',
  id: 123,
  node_id: 'string',
  avatar_url: 'string',
  gravatar_id: 'string',
  url: 'string',
  html_url: 'string',
  followers_url: 'string',
  following_url: 'string',
  gists_url: 'string',
  starred_url: 'string',
  subscriptions_url: 'string',
  organizations_url: 'string',
  repos_url: 'string',
  events_url: 'string',
  received_events_url: 'string',
  type: 'string',
  site_admin: false,
  score: 123,
}

describe('GithubUser', () => {
  it('matches snapshot', () => {
    const element = renderer.create(<GithubUser isLoading={false} user={mockUserFixture} />).toJSON();
    expect(element).toMatchSnapshot()
  })
})
