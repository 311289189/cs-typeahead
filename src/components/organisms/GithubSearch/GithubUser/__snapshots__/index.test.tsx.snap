// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubUser matches snapshot 1`] = `
<a
  className="userTile"
  href="string"
  rel="noopener noreferrer"
  target="_blank"
>
  <img
    alt="string"
    className="avatar"
    src="string"
  />
  <div
    className="userLogin"
  >
    string
  </div>
</a>
`;
